src/gradient.class.js

```diff
@@ -95,18 +95,59 @@
      */
     offsetY: 0,
 
+    /**
+     * A transform matrix to apply to the gradient before paiting.
+     * It shares the same name with the svg property but behaves differently.
+     * @type Array[Number]
+     * @default null
+     */
+    gradientTransform: null,
```

Comment:
1. painting
1. behaves differently how?  Describe the layout and the members of this matrix, please, enough to describe them for people coming along in the future.

```javascript
/**
 * | x1 x2 0 0 
 * | ... |
 * ...
 * x1: a description of x1
 * ...
 */
```

Yes, I know, JavaDoc is probably insufficient for this (I've never seen a 2-D array defined in JavaDoc in detail), but some documentation of the members is better than none.

Even better would be to make this an instance of an explicit unit-testable matrix class.  I realize that would be a significant refactor, but I'd be very surprised if you didn't have a need for it by now elsewhere in the code.  I do see there's a fabric.ColorMatrix class, but that appears to be something else.